Update for GNOME 48 and improve translations support

Woomymy opened the pull request:
Update for GNOME 45 support, update screenshots and format / cleanup some parts of the code.

GNOME 45 introduces breaking changes so older versions will no longer be supported

This should pass e.g.o guidelines this time. Thanks for your work on this extension, woomy

Woomymy committed:
metadata: Set version to 15 and restrict supported versions to GNOME 45

* Gnome 45 switched to ESM imports and it's therefore impossible to make
  our extension compatible with older shell versions

Signed-off-by: Woomymy <woomy@woomy.be>

Woomymy committed:
ideapad-controls: Update for GNOME 45 support

* Update imports to use ESM and move functions to Extension / ExtensionPreferences as required.

* Also refactor / cleanup code, lint with GNOME eslint rules and drop support for older GNOME versions (as GNOME 45 includes some breaking changes)

* Move TrayMenu and SystemMenu classes to menus.js

* See https://gjs.guide/extensions/upgrading/gnome-shell-45.html for
  more informations

Signed-off-by: Woomymy <woomy@woomy.be>

Woomymy committed:
images: Update screenshots

Signed-off-by: Woomymy <woomy@woomy.be>

Woomymy committed:
translations: Update POTFILES

Signed-off-by: Woomymy <woomy@woomy.be>

Woomymy committed:
translations: Use format to make translations better

* Should improve the quality of translations for some langages as
  requested in #13

Signed-off-by: Woomymy <woomy@woomy.be>

Woomymy commented:
The last commit is unrelated to GNOME 45 update but should improve the translations for some langages as requested in #13. I can seperate it in another PR if you want

AzzamAlsharafi commented:
Hello, 

First I want to thank you sir, you're the only person keeping this project alive. I have to apologize though, as it's probably going to take me a while to review this PR, as I'm currently very busy at the moment.

Regarding the translation improvement no need to separate to another PR.

Woomymy commented:
Hi, no problem if it takes you a while to review this and I'm happy to help if it can save you some time / help to keep this project alive. Thanks for precision for translation PR.

(also, to any user / dev reading this, if you want contribute by testing changes / giving some reviews, that would be welcome and very appreciated, thanks!)

4JX commented:
2021 Lenovo Legion 5 Pro (`16ach6h`) here, conservation/fn lock work fine.

My camera setup is borked somewhere and I don't have an USB charger to test with, so there's that.

Woomymy committed:
ideapad-controls: Update to version 16 for GNOME 46

* Tested on Fedora 40 beta with gnome 46, no other changes seem to be
  required

Signed-off-by: Woomymy <woomy@woomy.be>

josselinonduty reviewed (approved):
Support for Gnome 46 looks good to me

Woomymy committed:
ideapad-controls: Update to version 17 for GNOME 47

* Tested on Fedora 41 Beta with gnome 47, no other changes seem to be required

Signed-off-by: Woomymy <woomy@woomy.be>

josselinonduty commented:
Also, fixes #14

josselinonduty commented:
Fixes #14

@Woomymy you seem to be an active contributor. You should maybe fork and publish

ViBE-HU commented:
this extension is a very useful one. i am a nobody but would be good if someone could maintain it or fork it as a Lenovo Vantage alternative since the platform and the features are not Ideapad specific and Lenovo users have no Linux solution to control these settings which is a bummer..

Woomymy committed:
prefs: Move from GtkComboBoxText to GtkDropDown

* ComboBoxText is deprecated in 4.10

Signed-off-by: Woomymy <woomy@woomy.be>

Woomymy committed:
ideapad-controls: Update to version 18 for GNOME 48

* Tested on Fedora 42 Beta with gnome 48, no other changes seem to be required

Signed-off-by: Woomymy <woomy@woomy.be>

Woomymy committed:
makefile: Add extension_zip target

Signed-off-by: Woomymy <woomy@woomy.be>

Woomymy committed:
ideapad-controls: Add a script to create a mock sysfs dir

* Useful for VM tests

Signed-off-by: Woomymy <woomy@woomy.be>

Woomymy committed:
subprojects: Update blueprint-compiler

Signed-off-by: Woomymy <woomy@woomy.be>

Woomymy commented:
Hey @AzzamAlsharafi any updates ? Considering you seem quite busy (no problem, gl with life and stuff!), if you don't want / can't update upstream soon, I'll probably consider publishing a fork on gnome extensions website, don't hesitate to tell me if you see any issue with that

AzzamAlsharafi commented:
Hello @Woomymy, apologize for not contacting before. 

Yeah, I don't think I'll be coming back to this anytime soon, but I definitely support you publishing a fork on the gnome extensions site, or if there's a way for me to allow you to upload updates for the existing extension, or maybe transfer ownership, so that existing users can continue to receive updates without having to install a new extension.

Either way do tell me if you publish a fork, so that I can add a link to it in GitHub and the gnome extension page.

Thanks.

Woomymy committed:
ideapad-controls: Remove initTranslation call

* Deprecated in GNOME 45

Signed-off-by: Woomymy <woomy@woomy.be>

Woomymy committed:
ideapad-controls: Fork extension

Signed-off-by: Woomymy <woomy@woomy.be>

Woomymy committed:
ideapad-controls: Use notify() to send notifications even when pkexec is used

Signed-off-by: Woomymy <woomy@woomy.be>

Woomymy committed:
meta: Update to version 19

Signed-off-by: Woomymy <woomy@woomy.be>

Woomymy committed:
README: Update extension link

Signed-off-by: Woomymy <woomy@woomy.be>

Woomymy committed:
prefs: Gray out pkexec switch if tmpfiles config is not installed

Signed-off-by: Woomymy <woomy@woomy.be>

Woomymy committed:
meta: Update to version 20

Signed-off-by: Woomymy <woomy@woomy.be>

Woomymy commented:
Hey ! I published my fork to [e.g.o](https://extensions.gnome.org/extension/8007/ideapad-controls/), If you want to link it in the readme, don't hesitate to do it (either the repo or the whole extension).

Thanks again for your work on this extension, I'm always happy to help!

Woomymy commented:
@AzzamAlsharafi Concerning ownership, I don't know if you can't transfer it, but I would understand why it's not a good idea, anyway. I think just linking it in readme would be enough, and eventually on the GNOME extensions website (peaople searching would fine mine anyway)

Thanks again, woomy

albanobattistella committed:
Create it.po

Woomymy committed:
po: Fix missing quotes in it translation

Signed-off-by: Woomymy <woomy@woomy.be>

Woomymy committed:
Merge pull request #2 from albanobattistella/master

Create it.po

omeritzics committed:
Create he.po

Woomymy committed:
docs(README): Clarify tmpfiles-install instruction

Woomymy committed:
Merge pull request #4 from omeritzics/patch-1

Add Hebrew translations

Odaem committed:
meta: Update to GNOME 49

Seemingly no other changes necessary. Tested on Arch Linux (2025-09-27).

Odaem committed:
ideapad-controls: Fix 'Failed...' notification always sent when success notifications disabled

Fixes an issue where disabling 'send-success-notifications' would make
any option change send the "Failed to enable ..." notification.

Odaem committed:
template.ui: update and add dependency to Makefile

template.ui is meant to be generated from template.blp. Add this relationship to the makefile.

Odaem committed:
ideapad-controls: Fix fail notification never shown

Fixes an issue where the failure notification is not shown, even though
a write failure occured. The solution is to correctly handle the async
functions, helped by switching to the easier `Gio.Subprocess` API
instead of GLib.

Odaem committed:
ideapad-controls: Fix crash when sysfs-path is invalid

If sysfs-path was set by the user to an invalid path, the extension
could crash because of an uncaught exception. Fixed by wrapping in a
try-catch block.

Odaem committed:
make: Improve Makefile usability

* Add missing target dependencies.
* Be more verbose to show the user what is happening.
* Move 'all' target to the top to make it the default, following
  convention.

Odaem committed:
make: Change 'all' target to not install

Ensures Makefile follows convention: `make all` should only build, not
change the user's system by installing.

Woomymy committed:
Merge pull request #5 from Odaem/fixes

Update to GNOME 49 and various fixes

Woomymy committed:
chore: Bump version to 21

Woomymy committed:
chore: Update version to GNOME 50
